pandas/core/strings.py

```diff
@@ -623,13 +624,13 @@ def str_repeat(arr, repeats):
     dtype: object
     """
     if is_scalar(repeats):
-        def rep(x):
+        def scalar_rep(x):
```

Comment:
MyPy was throwing the following error with this:

```sh
pandas/core/strings.py:635: error: All conditional function variants must have identical signatures
```

I found a similar issue here:

https://github.com/python/mypy/issues/649

Generally I think this type of programming is going to be difficult to statically type check, so I think simply renaming this is an easy compromise to avoid tricking the static analysis